global_planner_plugins/a_star_planner/a_star_planner_gpp.cpp

```diff
@@ -12,49 +17,51 @@ PLUGINLIB_EXPORT_CLASS(cb_global_planner::AStarPlannerGPP, cb_global_planner::Gl
 
 // ----------------------------------------------------------------------------------------------------
 
-bool AStarPlannerGPP::queryEntityPose(const std::string& id, tf::Transform& pose)
+bool AStarPlannerGPP::queryEntityPose(const std::string& id, tf2::Transform& pose)
 {
     if (id == "map" || id == "/map")
     {
-        pose = tf::Transform::getIdentity();
+        pose = tf2::Transform::getIdentity();
         return true;
     }
 
     ed_msgs::SimpleQuery ed_query;
     ed_query.request.id = id;
 
-    std::cout << "[A* Planner] Querying ed for entity '" << id << "'." << std::endl;
+    ROS_DEBUG_STREAM("[A* Planner] Querying ed for entity '" << id << "'");
 
     if (!ed_client_.call(ed_query))
     {
-        ROS_ERROR_STREAM("[A* Planner] Failed to get pose for entity '" << id << "': ED could not be queried.");
+        ROS_ERROR_STREAM("[A* Planner] Failed to get pose for entity '" << id << "': ED could not be queried");
         return false;
     }
 
     if (ed_query.response.entities.empty())
     {
-        ROS_ERROR_STREAM("[A* Planner] Failed to get pose for entity '" << id << "': ED returns 'no such entity'.");
+        ROS_ERROR_STREAM("[A* Planner] Failed to get pose for entity '" << id << "': ED returns 'no such entity'");
         return false;
     }
 
     const ed_msgs::EntityInfo& e_info = ed_query.response.entities.front();
 
-    tf::poseMsgToTF(e_info.pose, pose);
+    tf2::fromMsg(e_info.pose, pose);
     return true;
 }
 
 // ----------------------------------------------------------------------------------------------------
 
-AStarPlannerGPP::AStarPlannerGPP() : global_costmap_ros_(NULL),  planner_(NULL) {}
+AStarPlannerGPP::AStarPlannerGPP() : global_costmap_ros_(nullptr), planner_(nullptr), tf_(nullptr)
+{
+}
 
-void AStarPlannerGPP::initialize(std::string name, tf::TransformListener* tf, costmap_2d::Costmap2DROS* global_costmap_ros)
+void AStarPlannerGPP::initialize(std::string /*name*/, tf2_ros::Buffer* tf, costmap_2d::Costmap2DROS* global_costmap_ros)
```

Comment:
Why is name commented out? does this not break stuff?

Reply:
No, this is common to do when a argument is not used, but it is not possible to change the API.